packages/manager/modules/pci/src/projects/project/storages/cold-archive/cold-archive.service.js

```diff
@@ -1,10 +1,147 @@
+import map from 'lodash/map';
 import 'moment';
 
 export default class PciStoragesColdArchiveService {
   /* @ngInject */
-  constructor($http, $q) {
+  constructor($http, $q, Poller, OvhApiCloudProjectUser) {
     this.$http = $http;
     this.$q = $q;
+    this.Poller = Poller;
+    this.OvhApiCloudProjectUser = OvhApiCloudProjectUser;
+  }
+
+  getUserDetails(projectId, userId) {
+    return this.$http
+      .get(`/cloud/project/${projectId}/user/${userId}`)
+      .then(({ data }) => data);
+  }
+
+  removeAllCredentials(projectId, userId, credentials) {
+    return this.$q.all(
+      map(credentials, (credential) =>
+        this.$http.delete(
+          `/cloud/project/${projectId}/user/${userId}/s3Credentials/${credential.access}`,
+        ),
+      ),
+    );
+  }
+
+  getS3Credentials(projectId, userId) {
+    return this.$http
+      .get(`/cloud/project/${projectId}/user/${userId}/s3Credentials`)
+      .then(({ data }) => data);
+  }
+
+  createUser(projectId, description, role) {
+    return this.$http
+      .post(`/cloud/project/${projectId}/user`, { description, role })
+      .then(({ data }) => data);
+  }
+
+  generateS3Credentials(projectId, userId) {
+    return this.$http
+      .post(`/cloud/project/${projectId}/user/${userId}/s3Credentials`)
+      .then(({ data }) => data);
+  }
+
+  getAllS3Users(projectId) {
+    return this.$http
+      .get(`/cloud/project/${projectId}/user`)
+      .then(({ data }) => data);
+  }
+
+  mapUsersToCredentials(projectId, users) {
+    const usersCredentialsPromises = users.map((user) =>
+      this.getS3Credentials(projectId, user.id)
+        .then((data) => ({
+          ...user,
+          s3Credentials: data,
+        }))
+        .catch(() => null),
+    );
+
+    return this.$q.all(usersCredentialsPromises);
+  }
+
+  getS3Users(projectId) {
+    this.OvhApiCloudProjectUser.v6().resetQueryCache();
+    return this.getAllS3Users(projectId).then((users) =>
+      this.getAndMapUsersHaveCredentials(projectId, users),
+    );
```

Comment:
**nitpick:** Better to use `$http`. We have seen cache issues with `ovh-api-services` library